### PR TITLE
Add a check for any pull failures

### PR DESCRIPTION
**Pull request by neehar18**
In reference to: https://github.com/redhat-developer/vscode-granite/issues/5

### CODE REVIEW

**Comment by fbricon** [comment]
please also handle error thrown by https://github.com/redhat-developer/vscode-granite/blob/main/src/panels/setupGranitePage.ts#L293

File: webviews/src/App.tsx [hunk 1]
```diff
@@ -99,10 +100,15 @@ function App() {
       case 'pull-progress': {
         const progress = payload.data.progress as ProgressData;
         const pulledModelName = progress.key;
+        const error = payload.data.error;
         //const progressData = data.progress as ProgressData;
         //find the ModelList component with the modelName value and update the progress
         //console.log(`Received pull-progress event ${pulledModelName} : ${progress}`);
         //console.log(`pulledModelName: ${pulledModelName} VS chatModel: ${chatModel}, tabModel: ${tabModel}, embeddingsModel: ${embeddingsModel}`);
+        if (error) {
+          setErrorMessage(`Error pulling model: ${error.message}`);
```
Comment (fbricon) [inline webviews/src/App.tsx:109]: the error message is never surfaced to the user

**Comment by fbricon** [comment]
I'll soon have a mock server ready, that will be usable to test various errors.

**Comment by fbricon** [comment]
Now with #43, you can throw some exceptions in
- https://github.com/redhat-developer/vscode-granite/blob/6c1a00d870e918fe17101b156bdd70bc511040e9/src/ollama/mockServer.ts#L175
- https://github.com/redhat-developer/vscode-granite/blob/main/src/ollama/mockServer.ts#L212

then you can check your changes

**Comment by neehar18** [comment]
Addressing this in [this](https://github.com/redhat-developer/vscode-granite/pull/61) PR. Closing this.

**Pull request closed without merging**